xbmc/cores/AudioEngine/Encoders/AEEncoderFFmpeg.cpp

```diff
@@ -36,9 +36,10 @@ CAEEncoderFFmpeg::CAEEncoderFFmpeg() : m_CodecCtx(NULL), m_SwrCtx(NULL)
 CAEEncoderFFmpeg::~CAEEncoderFFmpeg()
 {
   Reset();
-  swr_free(&m_SwrCtx);
-  av_channel_layout_uninit(&m_CodecCtx->ch_layout);
-  avcodec_free_context(&m_CodecCtx);
+  if (m_SwrCtx)
+    swr_free(&m_SwrCtx);
+  if (m_CodecCtx)
+    avcodec_free_context(&m_CodecCtx);
```

Comment:
`swr_free` and `avcodec_free_context` check themself if they are already null.

https://github.com/FFmpeg/FFmpeg/blob/722a2170e83231283fc74bede495b3b4ee9591ac/libswresample/swresample.c#L121

https://github.com/FFmpeg/FFmpeg/blob/722a2170e83231283fc74bede495b3b4ee9591ac/libavcodec/options.c#L164

Reply:
Interesting, I've removed the checks. `av_channel_layout_uninit` was causing the issue as the access to the member variable is unsafe as `m_CodecCtx` can be null. I've changed the commit. The whole encoder class could use an overhaul. Half of the code is completely unused (like m_SwrCtx which is set but never used).